### PR TITLE
NDEV-1569, NDEV-1611: get account_at_slot,  switch neon-cli to ClickHouse usage

### PR DESCRIPTION
**Pull request by valiksinev**
need to run performance test

### CODE REVIEW

File: evm_loader/cli/src/types/tracer_ch_db.rs [hunk 1]
```diff
@@ -36,9 +64,22 @@ impl ClickHouseDb {
...
+                .await
+                .map_err(std::convert::Into::into)
+        })
+    }
+
     pub fn get_block_time(&self, slot: Slot) -> ChResult<UnixTimestamp> {
         block(|| async {
-            let query = "SELECT JSONExtractInt(notify_block_json, 'block_time') FROM events.notify_block_local WHERE (slot = toUInt64(?))";
+            let query =
+                "SELECT JSONExtractInt(notify_block_json, 'block_time') FROM events.notify_block_local WHERE (slot = toUInt64(?))";
```
Comment (Deniskore) [inline evm_loader/cli/src/types/tracer_ch_db.rs:82]: I think we can remove toUint64() here, but we need to check it.

Comment (valiksinev) [reply evm_loader/cli/src/types/tracer_ch_db.rs:82]: removed

File: evm_loader/cli/src/types/tracer_ch_db.rs [hunk 2]
```diff
@@ -59,4 +100,193 @@ impl ClickHouseDb {
...
+                .query(query)
+                .fetch_one::<u64>()
+                .await
+                .map_err(std::convert::Into::into)
+        })
+    }
+
+    fn get_branch_slots(&self, slot: u64) -> ChResult<(u64, Vec<u64>)> {
+        let rows: Vec<SlotParent> = block(|| async {
+            let query = "SELECT distinct on slot, ?fields FROM events.update_slot \
```
[the remark this answers is not in the document]
Comment (valiksinev) [reply evm_loader/cli/src/types/tracer_ch_db.rs:117]: ?fields should be replaced by the field names from the SlotParent struct.

File: evm_loader/cli/src/types/tracer_ch_db.rs [hunk 3]
```diff
@@ -59,4 +100,193 @@ impl ClickHouseDb {
...
+
+        let (root, rows) = rows.split_last().ok_or_else(|| {
+            let err = clickhouse::error::Error::Custom("Rooted slot not found".to_string());
+            ChError::Db(err)
+        })?;
+
+        match slot.cmp(&root.slot) {
+            Less => {
+                let count = block(|| async {
+                    let query = "SELECT count(*) FROM events.update_slot WHERE slot = ? ands status = 'Rooted'";
```
[the remark this answers is not in the document]
Comment (valiksinev) [reply evm_loader/cli/src/types/tracer_ch_db.rs:132]: fixed

File: evm_loader/cli/src/types/tracer_ch_db.rs [hunk 4]
```diff
@@ -59,4 +100,193 @@ impl ClickHouseDb {
...
+                Err(e) => return Err(ChError::Db(e)),
+            };
+        }
+
+        if row.is_none() {
+            let result = block(|| async {
+                let query = r#"
+                SELECT owner, lamports, executable, rent_epoch, data
+                FROM events.older_account_distributed
+                WHERE pubkey = ?
```
[the remark this answers is not in the document]
Comment (valiksinev) [reply evm_loader/cli/src/types/tracer_ch_db.rs:245]: The older_account table has  PRIMARY KEY (pubkey).
If you are right, the primary key must be changed to 
 PRIMARY KEY (pubkey, slot, retrieved_time)

[the remark this answers is not in the document]
Comment (ivandzen) [reply evm_loader/cli/src/types/tracer_ch_db.rs:247]: Primary key does not guarantee deduplication in clickhouse. Please, fix query

File: evm_loader/cli/src/types/tracer_ch_db.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivandzen) [reply evm_loader/cli/src/types/tracer_ch_db.rs:117]: Anyway, syntax of query is wrong DISTINCT ON (slot)

[the remark this answers is not in the document]
Comment (valiksinev) [reply evm_loader/cli/src/types/tracer_ch_db.rs:117]: fixed

File: evm_loader/cli/src/types/tracer_ch_db.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (valiksinev) [reply evm_loader/cli/src/types/tracer_ch_db.rs:247]: fixed

**Review by ivandzen — APPROVED** [review]

File: evm_loader/cli/src/types/tracer_ch_db.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (BoogVAr) [reply evm_loader/cli/src/types/tracer_ch_db.rs:249]: Please use key work final. It's native way of working with tables like that. The ordering will be more heavy than finalisasion.

File: evm_loader/cli/src/types/tracer_ch_db.rs [hunk 5]
```diff
@@ -15,18 +26,37 @@ pub enum ChError {
...
 }
 
 #[allow(dead_code)]
 impl ClickHouseDb {
-    pub fn _new(server_url: &str, username: Option<&str>, password: Option<&str>) -> ClickHouseDb {
-        let client = match (username, password) {
-            (None, None | Some(_)) => Client::default().with_url(server_url),
-            (Some(user), None) => Client::default().with_url(server_url).with_user(user),
+    pub fn new(config: &ChDbConfig) -> Self {
+        println!("{:?}", config);
```
Comment (Deniskore) [inline evm_loader/cli/src/types/tracer_ch_db.rs:52]: Please remove this println!

**Review by Deniskore — CHANGES_REQUESTED** [review]

File: evm_loader/cli/src/types/tracer_ch_db.rs (hunk 5 above)
Comment (valiksinev) [reply evm_loader/cli/src/types/tracer_ch_db.rs:52]: fixed

**Pull request merged**